sdk/storage/inc/shares/share_responses.hpp

```diff
@@ -26,4 +27,93 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
   using GetShareAccessPolicyResult = ShareGetAccessPolicyResult;
   using GetSharePermissionResult = ShareGetPermissionResult;
 
+  // DirectoryClient models:
+  using CreateDirectoryResult = DirectoryCreateResult;
+  using DeleteDirectoryResult = DirectoryDeleteResult;
+  using GetDirectoryPropertiesResult = DirectoryGetPropertiesResult;
+  using SetDirectoryPropertiesResult = DirectorySetPropertiesResult;
+  using SetDirectoryMetadataResult = DirectorySetMetadataResult;
+  using ForceCloseDirectoryHandlesResult = DirectoryForceCloseHandlesResult;
+
+  struct ListFilesAndDirectoriesSegmentedResult
+  {
+    std::string ServiceEndpoint;
+    std::string ShareName;
+    std::string ShareSnapshot;
+    std::string DirectoryPath;
+    std::string Prefix;
+    std::string Marker;
+    int32_t MaxResults = int32_t();
+    std::string NextMarker;
+    std::vector<DirectoryItem> DirectoryItems;
+    std::vector<FileItem> FileItems;
+  };
+
+  struct ListDirectoryHandlesSegmentedResult
+  {
+    std::vector<HandleItem> HandleList;
+    std::string NextMarker;
+  };
+
+  struct FileShareSmbProperties
+  {
+    /**
+     * @brief Key of the permission to be set for the directory/file.
+     */
+    std::string FilePermissionKey;
+
+    /**
+     * @brief If specified, the provided file attributes shall be set. Default value:
+     * FileAttribute::Archive for file and FileAttribute::Directory for directory.
+     * FileAttribute::None can also be specified as default.
+     */
+    FileAttributes Attributes = static_cast<FileAttributes>(0);
+
+    /**
+     * @brief Creation time for the file/directory. Default value: Now.
+     */
+    std::string FileCreationTime = std::string("now");
+
+    /**
+     * @brief Last write time for the file/directory. Default value: Now.
+     */
+    std::string FileLastWriteTime = std::string("now");
+  };
+
+  // FileClient models:
+  using CreateFileResult = FileCreateResult;
+  using DeleteFileResult = FileDeleteResult;
+  using DownloadFileResult = FileDownloadResult;
+  using StartCopyFileResult = FileStartCopyResult;
+  using AbortCopyFileResult = FileAbortCopyResult;
+  using GetFilePropertiesResult = FileGetPropertiesResult;
+  using SetFilePropertiesResult = FileSetHTTPHeadersResult;
+  using ResizeFileResult = FileSetHTTPHeadersResult;
+  using SetFileMetadataResult = FileSetMetadataResult;
+  using UploadFileRangeResult = FileUploadRangeResult;
+  using ClearFileRangeResult = FileUploadRangeResult;
+  using UploadFileRangeFromUrlResult = FileUploadRangeFromURLResult;
+  using GetFileRangeListResult = FileGetRangeListResult;
+  using ListFileHandlesSegmentedResult = ListDirectoryHandlesSegmentedResult;
+  using ForceCloseFileHandlesResult = FileForceCloseHandlesResult;
+  using AcquireFileLeaseResult = FileAcquireLeaseResult;
+  using ReleaseFileLeaseResult = FileReleaseLeaseResult;
+  using BreakFileLeaseResult = FileBreakLeaseResult;
+  using ChangeFileLeaseResult = FileChangeLeaseResult;
```

Comment:
I feel terrible about these meaningless typedefs.

Reply:
To align with the naming convention.